lib/clashom.gi

```diff
@@ -2362,11 +2362,23 @@ local r,	#radical
   # factor out the radical first. (Note: The radical method for perm groups
   # stores the nat hom.!)
   ser:=FittingFreeLiftSetup(G);
-  radsize:=Product(RelativeOrders(ser.pcgs));
+  if Length(ser.pcgs)>0 then
+    radsize:=Product(RelativeOrders(ser.pcgs));
+  else
+    radsize:=1;
+  fi;
   len:=Length(ser.pcgs);
 
   if radsize=1 then
-    cl:=ConjugacyClassesFittingFreeGroup(G:onlysizes:=false);
+    hom:=ser.factorhom;
+    if IsPermGroup(Range(hom)) and not IsPermGroup(Source(hom)) then
+      f:=Image(hom,G);
+      cl:=ConjugacyClassesFittingFreeGroup(f:onlysizes:=false);
+      cl:=List(cl,x->[PreImagesRepresentative(hom,x[1]),
+        PreImage(hom,x[2])]);
```

Comment:
This new code is not covered by a test; do you perhaps have a simple test case?

Reply:
Alas, these all came up in a large example, and I did not bother to construct a smaller error trigger.